apps/web/app/modules/Admin/AddCourse/AddCourse.tsx

```diff
@@ -0,0 +1,190 @@
+import { useNavigate } from "@remix-run/react";
+import { useCallback, useState } from "react";
+
+import { useUploadFile } from "~/api/mutations/admin/useUploadFile";
+import { useCategoriesSuspense } from "~/api/queries/useCategories";
+import SplashScreenImage from "~/assets/splash-screen-image.svg";
+import ImageUploadInput from "~/components/FileUploadInput/ImageUploadInput";
+import { Icon } from "~/components/Icon";
+import Editor from "~/components/RichText/Editor";
+import { Button } from "~/components/ui/button";
+import { Form, FormControl, FormField, FormItem, FormMessage } from "~/components/ui/form";
+import { Input } from "~/components/ui/input";
+import { Label } from "~/components/ui/label";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "~/components/ui/select";
+
+import { useAddCourseForm } from "./hooks/useAddCourseForm";
+
+const AddCourse = () => {
+  const navigate = useNavigate();
+  const { form, onSubmit } = useAddCourseForm();
+  const { data: categories } = useCategoriesSuspense();
+  const [isUploading, setIsUploading] = useState(false);
+  const { mutateAsync: uploadFile } = useUploadFile();
+  const { isValid: isFormValid } = form.formState;
+  const watchedImageUrl = form.watch("imageUrl");
+  const imageUrl = form.getValues("imageUrl");
+
+  const handleImageUpload = useCallback(
+    async (file: File) => {
+      setIsUploading(true);
+      try {
+        const result = await uploadFile({ file, resource: "course" });
+        form.setValue("imageUrl", result.fileUrl, { shouldValidate: true });
+      } catch (error) {
+        console.error("Error uploading image:", error);
+      } finally {
+        setIsUploading(false);
+      }
+    },
+    [form],
+  );
+
+  return (
+    <div className="flex h-screen bg-white">
+      <div className="w-1/2 flex items-center justify-center">
+        <img src={SplashScreenImage} alt="splashScreenImage" className="rounded" />
+      </div>
+      <div className="w-1/2 mt-[1rem]">
+        <Button
+          onClick={() => navigate("/admin/courses")}
+          className="border-2 border-beige-500 bg-white text-blue-500 py-2 px-6 rounded"
+        >
+          <Icon name="ChevronLeft" className="w-3 h-3 mr-2" />
+          Back
+        </Button>
+        <h1 className="text-3xl font-bold mt-[2rem]">Create new course</h1>
+        <h1 className="mt-5">
+          Provide the details to set up a new course. You’ll have more options to customize it
+          later.
+        </h1>
+        <Form {...form}>
+          <form className="mt-[2.5rem]" onSubmit={form.handleSubmit(onSubmit)}>
```

Comment:
Same case as above 😉 
```suggestion
          <form className="mt-10" onSubmit={form.handleSubmit(onSubmit)}>
```